device/src/bin/v2.rs

```diff
@@ -109,13 +112,14 @@ fn main() -> ! {
     let spi = Spi::new(peripherals.SPI2, 40u32.MHz(), SpiMode::Mode2, &clocks)
         .with_sck(io.pins.gpio8)
         .with_mosi(io.pins.gpio7);
-    let di = SPIInterfaceNoCS::new(spi, io.pins.gpio9.into_push_pull_output());
-    let display = Builder::st7789(di)
-        .with_display_size(240, 280)
-        .with_window_offset_handler(|_| (0, 20)) // 240*280 panel
-        .with_invert_colors(ColorInversion::Inverted)
-        .with_orientation(Orientation::Portrait(false))
-        .init(&mut delay, Some(io.pins.gpio6.into_push_pull_output())) // RES
+    let spi_device = embedded_hal_bus::spi::ExclusiveDevice::new_no_delay(spi, NoCs);
+    let di = SPIInterface::new(spi_device, Output::new(io.pins.gpio9, Level::Low));
```

Comment:
It'd be interesting to know what the `embedded_hal_bus` crate is doing that the original Spi type was not. It sounds like from the documentation it helps different devices share the bus. Are we doing that?

Reply:
Nope, it's just for the display

Reply:
Right so why do we need the `embedded_hal_bus` crate. What goes wrong without it?

Reply:
display_interface_spi expects that SpiDevice embedded-hal-bus type now